Indicators/RelativeDailyVolume.cs

```diff
@@ -0,0 +1,145 @@
+        public RelativeDailyVolume(int period = 2, Resolution resolution = Resolution.Daily)
+            : this($"RDV({period})", period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new RelativeDailyVolume indicator with the specified period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        public RelativeDailyVolume(string name, int period)
+            : base(name)
+        {
+            _relativeData = new Dictionary<DateTime, SimpleMovingAverage>();
+            _currentData = new Dictionary<DateTime, decimal>();
+            WarmUpPeriod = period;
+            _previousDay = -1; /// No calendar day can be -1, thus default is not a calendar day
+            _days = -1; /// Will increment by one after first TradeBar, then will increment by one every new day
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A a value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            if (input.Time.Day != _previousDay)
+            {
+                var cummulativeVolume = 0;
+                foreach (var pair in _currentData)
+                {
+                    /// Arbitrary year, month, day
+                    DateTime timeBar = new DateTime(1, 1, 1, pair.Key.Hour, pair.Key.Minute, pair.Key.Second);
+                    if (!_relativeData.ContainsKey(timeBar))
+                    {
+                        _relativeData[timeBar] = new SimpleMovingAverage(WarmUpPeriod);
+                    }
+                    cummulativeVolume += (int)pair.Value;
+                    _relativeData[timeBar].Update(pair.Key, cummulativeVolume);
+                }
+                _currentData.Clear();
+                _previousDay = input.Time.Day;
+                _days += 1;
+            }
+
+            if (!_currentData.ContainsKey(input.Time))
+            {
+                _currentData[input.Time] = input.Volume;
+            }
+
+            if (!IsReady)
+            {
+                return 0;
+            }
+
+            /// Arbitrary year, month, day
+            DateTime currentTimeBar = new DateTime(1, 1, 1, input.Time.Hour, input.Time.Minute, input.Time.Second);
+            var denominator = 0.0m;
+            if (!_relativeData.ContainsKey(currentTimeBar))
+            {
+                /// If there is no historical data for the current time, get most recent historical data
+                /// This may come into play for crypto assets or a circuit breaker event
+                var relativeDataKeys = _relativeData.Keys.ToList();
+                relativeDataKeys.Sort((x, y) => DateTime.Compare(x, y));
```

Comment:
Can avoid the need for this sorting directly using a sorted dictionary and going through it?